File: data/compilers_comparison/src/examples/second_example/src/func2.c
#include "second_example.h" 
array_t* func2(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   array_t* array247;
   if (pCounter > 0) {
      array247 = vars->data[--pCounter];
      array247->refC++;
      DEBUG_COPY(array247->id);
   } else {
      array247 = (array_t*)malloc(sizeof(array_t));
      array247->size = 30;
      array247->refC = 1;
      array247->id = 247;
      array247->data = (unsigned int*)malloc(array247->size*sizeof(unsigned int));
      memset(array247->data, 0, array247->size*sizeof(unsigned int));
      DEBUG_NEW(array247->id);
   }
   unsigned int loop71 = 0;
   unsigned int loopLimit71 = (rand()%loopsFactor)/2 + 1;
   for(; loop71 < loopLimit71; loop71++) {
      if(PATH0 & 1) {
         array_t* array248;
         if (pCounter > 0) {
            array248 = vars->data[--pCounter];
            array248->refC++;
            DEBUG_COPY(array248->id);
         } else {
            array248 = (array_t*)malloc(sizeof(array_t));
            array248->size = 573;
            array248->refC = 1;
            array248->id = 248;
            array248->data = (unsigned int*)malloc(array248->size*sizeof(unsigned int));
            memset(array248->data, 0, array248->size*sizeof(unsigned int));
            DEBUG_NEW(array248->id);
         }
         unsigned int loop72 = 0;
         unsigned int loopLimit72 = (rand()%loopsFactor)/3 + 1;
         for(; loop72 < loopLimit72; loop72++) {
            for (int i = 0; i < array247->size; i++) {
               array247->data[i]--;
            }
            for (int i = 0; i < array248->size; i++) {
               if (array248->data[i] == 22) { 
                  return array248;
               }
            }
         }
         array_t_param params0;
         params0.size = 2;
         params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
         params0.data[0] = array247;
         params0.data[1] = array248;
         array_t* array249 = func18(&params0, rng(), loopsFactor);
         DEBUG_RETURN(array249->id);
         free(params0.data);
         array249->refC--;
         if(array249->refC == 0) {
            free(array249->data);
            free(array249);
            DEBUG_FREE(array249->id);
         }
         array248->refC--;
         if(array248->refC == 0) {
            free(array248->data);
            free(array248);
            DEBUG_FREE(array248->id);
         }
      }
      else {
         array_t_param params0;
         params0.size = 1;
         params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
         params0.data[0] = array247;
         array_t* array250 = func14(&params0, rng(), loopsFactor);
         DEBUG_RETURN(array250->id);
         free(params0.data);
         array250->refC--;
         if(array250->refC == 0) {
            free(array250->data);
            free(array250);
            DEBUG_FREE(array250->id);
         }
      }
      array_t* array251;
      if (pCounter > 0) {
         array251 = vars->data[--pCounter];
         array251->refC++;
         DEBUG_COPY(array251->id);
      } else {
         array251 = (array_t*)malloc(sizeof(array_t));
         array251->size = 232;
         array251->refC = 1;
         array251->id = 251;
         array251->data = (unsigned int*)malloc(array251->size*sizeof(unsigned int));
         memset(array251->data, 0, array251->size*sizeof(unsigned int));
         DEBUG_NEW(array251->id);
      }
      array_t_param params0;
      params0.size = 2;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      params0.data[0] = array247;
      params0.data[1] = array251;
      array_t* array252 = func15(&params0, loopsFactor);
      DEBUG_RETURN(array252->id);
      free(params0.data);
      if(PATH0 & 2) {
         array_t* array253;
         if (pCounter > 0) {
            array253 = vars->data[--pCounter];
            array253->refC++;
            DEBUG_COPY(array253->id);
         } else {
            array253 = (array_t*)malloc(sizeof(array_t));
            array253->size = 443;
            array253->refC = 1;
            array253->id = 253;
            array253->data = (unsigned int*)malloc(array253->size*sizeof(unsigned int));
            memset(array253->data, 0, array253->size*sizeof(unsigned int));
            DEBUG_NEW(array253->id);
         }
         unsigned int loop73 = 0;
         unsigned int loopLimit73 = (rand()%loopsFactor)/3 + 1;
         for(; loop73 < loopLimit73; loop73++) {
            for (int i = 0; i < array252->size; i++) {
               array252->data[i]--;
            }
            for (int i = 0; i < array252->size; i++) {
               if (array252->data[i] == 40) { 
                  return array252;
               }
            }
         }
         array_t_param params1;
         params1.size = 4;
         params1.data = (array_t**)malloc(params1.size*sizeof(array_t*));
         params1.data[0] = array247;
         params1.data[1] = array251;
         params1.data[2] = array252;
         params1.data[3] = array253;
         array_t* array254 = func24(&params1, rng(), loopsFactor);
         DEBUG_RETURN(array254->id);
         free(params1.data);
         array254->refC--;
         if(array254->refC == 0) {
            free(array254->data);
            free(array254);
            DEBUG_FREE(array254->id);
         }
         array253->refC--;
         if(array253->refC == 0) {
            free(array253->data);
            free(array253);
            DEBUG_FREE(array253->id);
         }
      }
      else {
         array_t_param params1;
         params1.size = 3;
         params1.data = (array_t**)malloc(params1.size*sizeof(array_t*));
         params1.data[0] = array247;
         params1.data[1] = array251;
         params1.data[2] = array252;
         array_t* array255 = func16(&params1, rng(), loopsFactor);
         DEBUG_RETURN(array255->id);
         free(params1.data);
         array255->refC--;
         if(array255->refC == 0) {
            free(array255->data);
            free(array255);
            DEBUG_FREE(array255->id);
         }
      }
      for (int i = 0; i < array247->size; i++) {
         if (array247->data[i] == 41) { 
            return array247;
         }
      }
      array252->refC--;
      if(array252->refC == 0) {
         free(array252->data);
         free(array252);
         DEBUG_FREE(array252->id);
      }
      array251->refC--;
      if(array251->refC == 0) {
         free(array251->data);
         free(array251);
         DEBUG_FREE(array251->id);
      }
   }
   for (int i = 0; i < array247->size; i++) {
      array247->data[i]--;
   }
   for (int i = 0; i < array247->size; i++) {
      if (array247->data[i] == 24) { 
         return array247;
      }
   }
   return array247;
}

File: data/multilang_comparison/program_go/program_go/src/func2.go
package main
import "math/rand"
var _ = rand.Intn
func Func2(vars *ArrayParam, PATH0 uint64 , loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   var array215 *Array
   if pCounter > 0 {
       array215 = vars.Data[pCounter-1]
       pCounter--
       array215.RefC++
   } else {
       array215 = &Array{
           Size: 993,
           RefC: 1,
           Id: 215,
           Data: make([]uint32, 993),
       }
   }
   var loop62 int = 0;
   var loopLimit62 int = (50)/2 + 1;
   for ; loop62 < loopLimit62; loop62++ {
      if (PATH0 & 1) != 0 {
         var array216 *Array
         if pCounter > 0 {
             array216 = vars.Data[pCounter-1]
             pCounter--
             array216.RefC++
         } else {
             array216 = &Array{
                 Size: 918,
                 RefC: 1,
                 Id: 216,
                 Data: make([]uint32, 918),
             }
         }
         var loop63 int = 0;
         var loopLimit63 int = (50)/3 + 1;
         for ; loop63 < loopLimit63; loop63++ {
            for i := 0; i < array216.Size; i++ {
                array216.Data[i]--
            }
            for i := 0; i < array215.Size; i++ {
                if array215.Data[i] == 22 {
                    return array215
                }
            }
         }
         var params0 ArrayParam
         params0.Size = 2
         params0.Data = make([]*Array, params0.Size)
         params0.Data[0] = array215
         params0.Data[1] = array216
         var array217 *Array = Func18(&params0, getPath(), loopsFactor);
         params0.Data = nil
         params0.Size = 0
         array217.RefC--
         if array217.RefC == 0 {
             array217.Data = nil
             array217 = nil
         }
         array216.RefC--
         if array216.RefC == 0 {
             array216.Data = nil
             array216 = nil
         }
      
      } else {
         var params0 ArrayParam
         params0.Size = 1
         params0.Data = make([]*Array, params0.Size)
         params0.Data[0] = array215
         var array218 *Array = Func14(&params0, getPath(), loopsFactor);
         params0.Data = nil
         params0.Size = 0
         array218.RefC--
         if array218.RefC == 0 {
             array218.Data = nil
             array218 = nil
         }
      }
      var array219 *Array
      if pCounter > 0 {
          array219 = vars.Data[pCounter-1]
          pCounter--
          array219.RefC++
      } else {
          array219 = &Array{
              Size: 574,
              RefC: 1,
              Id: 219,
              Data: make([]uint32, 574),
          }
      }
      var params0 ArrayParam
      params0.Size = 2
      params0.Data = make([]*Array, params0.Size)
      params0.Data[0] = array215
      params0.Data[1] = array219
      var array220 *Array = Func15(&params0, loopsFactor);
      params0.Data = nil
      params0.Size = 0
      if (PATH0 & 2) != 0 {
         var array221 *Array
         if pCounter > 0 {
             array221 = vars.Data[pCounter-1]
             pCounter--
             array221.RefC++
         } else {
             array221 = &Array{
                 Size: 486,
                 RefC: 1,
                 Id: 221,
                 Data: make([]uint32, 486),
             }
         }
         var loop64 int = 0;
         var loopLimit64 int = (50)/3 + 1;
         for ; loop64 < loopLimit64; loop64++ {
            for i := 0; i < array219.Size; i++ {
                array219.Data[i]--
            }
            for i := 0; i < array221.Size; i++ {
                if array221.Data[i] == 74 {
                    return array221
                }
            }
         }
         var params1 ArrayParam
         params1.Size = 4
         params1.Data = make([]*Array, params1.Size)
         params1.Data[0] = array215
         params1.Data[1] = array219
         params1.Data[2] = array220
         params1.Data[3] = array221
         var array222 *Array = Func24(&params1, getPath(), loopsFactor);
         params1.Data = nil
         params1.Size = 0
         array222.RefC--
         if array222.RefC == 0 {
             array222.Data = nil
             array222 = nil
         }
         array221.RefC--
         if array221.RefC == 0 {
             array221.Data = nil
             array221 = nil
         }
      
      } else {
         var params1 ArrayParam
         params1.Size = 3
         params1.Data = make([]*Array, params1.Size)
         params1.Data[0] = array215
         params1.Data[1] = array219
         params1.Data[2] = array220
         var array223 *Array = Func16(&params1, getPath(), loopsFactor);
         params1.Data = nil
         params1.Size = 0
         array223.RefC--
         if array223.RefC == 0 {
             array223.Data = nil
             array223 = nil
         }
      }
      for i := 0; i < array215.Size; i++ {
          if array215.Data[i] == 75 {
              return array215
          }
      }
      array220.RefC--
      if array220.RefC == 0 {
          array220.Data = nil
          array220 = nil
      }
      array219.RefC--
      if array219.RefC == 0 {
          array219.Data = nil
          array219 = nil
      }
   }
   for i := 0; i < array215.Size; i++ {
       array215.Data[i]--
   }
   for i := 0; i < array215.Size; i++ {
       if array215.Data[i] == 10 {
           return array215
       }
   }
   return array215;
}

